src/java/simpledb/TupleDesc.java

```diff
@@ -122,16 +140,45 @@ public Type getFieldType(int i) throws NoSuchElementException {
      */
     public int fieldNameToIndex(String name) throws NoSuchElementException {
         // some code goes here
-        return 0;
+        int i,num;
+        if(name==null)
+            throw new NoSuchElementException();
+        boolean nameval= false;
+        try{
+        for(i=0;i<tdlist.size();i++)
```

Comment:
Again don't extend variable scopes as long as they aren't needed! Stick to `for(int i = 0; ...`